sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/DistinctHash.java

```diff
@@ -3,60 +3,130 @@
 package com.azure.cosmos.implementation.query;
 
 import com.azure.cosmos.implementation.JsonSerializable;
-import com.azure.cosmos.implementation.Utils;
 import com.azure.cosmos.implementation.routing.MurmurHash3_128;
 import com.azure.cosmos.implementation.routing.UInt128;
-import com.azure.cosmos.models.ModelBridgeInternal;
-import com.fasterxml.jackson.databind.MapperFeature;
-import com.fasterxml.jackson.databind.ObjectMapper;
-import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.ValueNode;
 
 import java.io.IOException;
-import java.nio.ByteBuffer;
+import java.util.Iterator;
 import java.util.List;
+import java.util.Map;
 
 public final class DistinctHash {
 
-    private static final UInt128 ARRAY_HASH_SEED = new UInt128(0xfa573b014c4dc18eL, 0xa014512c858eb115L);
+    private static final UInt128 RootHashSeed = new UInt128(0xbfc2359eafc0e2b7L, 0x8846e00284c4cf1fL);
 
-    private static final ObjectMapper OBJECT_MAPPER =
-        new ObjectMapper()
-            .configure(MapperFeature.SORT_PROPERTIES_ALPHABETICALLY, true)
-            .configure(SerializationFeature.ORDER_MAP_ENTRIES_BY_KEYS, true);
+    private static class HashSeeds
+    {
```

Comment:
code style: "{" on the same line as class definition.

Reply:
updated